pandaserd/module.py

```diff
@@ -196,26 +205,30 @@ def create_rel(self, left_table_name, right_table_name, left_on=None, right_on=N
             self.table_gen_code.append(rel)
 
 
-    def write_to_file(self, filename='output.txt'):
+    def write_to_file(self, filename: str = 'output.dot'):
         """
         Encloses the current tables and relationships into a DiGraph object (dot code) and
         writes output dot code to a text file.
         :param filename: (str) file to output the dot code to
         """
-        self.filename = filename
-
         # did the code already end before?
         if '\t}' in set(self.table_gen_code):
             self.table_gen_code.remove('\t}')
-
         self.table_gen_code.append('\t}')
-
         tmp = self.table_gen_code
         self.res = '\n'.join(tmp)
 
-        text_file = open(self.filename, "w")
-        text_file.write(self.res)
-        text_file.close()
-
-        url='https://edotor.net/'
-        print(f'written to {self.filename}; visit {url} to render ERD')
+        path = Path(filename)
+        self.filename = path.with_suffix('.dot')
+        dot_file = open(self.filename.as_posix(), "w")
+        dot_file.write(self.res)
+        dot_file.close()
+
+        if path.suffix != '.dot':
+            src = Source.from_file(self.filename.as_posix())
+            fmt = path.suffix.replace('.', '')
+            src.render(path.with_suffix(''), format=fmt)
+            print(f'image written to {filename}')
+        else:
+            url='https://edotor.net/'
```

Comment:
Perhaps add `http://magjac.com/graphviz-visual-editor/` as an option as well? Given that Edotor does not render all the HTML colors from the list.